internal/host/validator.go

```diff
@@ -342,6 +343,67 @@ func (v *validator) printCompatibleWithClusterPlatform(c *validationContext, sta
 	}
 }
 
+func isDiskEncryptionRelevantForHost(c *validationContext) bool {
+	switch *c.cluster.DiskEncryption.EnableOn {
+	case models.DiskEncryptionEnableOnAll:
+		return true
+	case models.DiskEncryptionEnableOnMasters:
+		return c.host.Role == models.HostRoleMaster || c.host.Role == models.HostRoleBootstrap || c.host.Role == models.HostRoleAutoAssign
+	case models.DiskEncryptionEnableOnWorkers:
+		return c.host.Role == models.HostRoleWorker || c.host.Role == models.HostRoleAutoAssign
+	default: // models.DiskEncryptionEnableOnNone
+		return false
+	}
+}
+
+func isAutoassignAllowed(c *validationContext) bool {
+	return *c.cluster.DiskEncryption.EnableOn == models.DiskEncryptionEnableOnAll ||
+		*c.cluster.DiskEncryption.EnableOn == models.DiskEncryptionEnableOnNone
+}
+
+func (v *validator) diskEncryptionRequirementsSatisfied(c *validationContext) ValidationStatus {
+
+	if c.infraEnv != nil || reflect.DeepEqual(c.cluster.DiskEncryption, &models.DiskEncryption{}) || !isDiskEncryptionRelevantForHost(c) {
+		return ValidationSuccessSuppressOutput
+	}
+
+	if c.host.Role == models.HostRoleAutoAssign && !isAutoassignAllowed(c) {
```

Comment:
After role suggestion is implemented, this should be always Pending

Reply:
Will the role be set before the host goes to `known` state? If yes, can you please make this change part of your [PR](https://github.com/openshift/assisted-service/pull/2541) ?